_posts/2024-01-30-securing-wildfly-apps-oidc-k8s.adoc

```diff
@@ -0,0 +1,398 @@
+                        </execution>
+                    </executions>
+                </plugin>
+            </plugins>
+        </build>
+    </profile>
+</profiles>
+----
+<1> *wildfly-maven-plugin* provisions a WildFly server with the specified layers with our application deployed.
+<2> *elytron-oidc-client* automatically adds the native OIDC client subsystem to our WildFly installation.
+
+* Examine the https://github.com/wildfly-security-incubator/elytron-examples/blob/main/simple-webapp-oidc/src/main/webapp/WEB-INF/web.xml[web.xml].
++
+[source,xml]
+----
+...
+    <login-config>
+        <auth-method>OIDC</auth-method>  <1>
+    </login-config>
+...
+----
+<1> When *elytron-oidc-client* subsystem sees *auth-method* is set to *OIDC*, it enables OIDC authentication mechanism for the application.
+
+* Examine the https://github.com/wildfly-security-incubator/elytron-examples/blob/main/simple-webapp-oidc/src/main/webapp/WEB-INF/oidc.json[oidc.json] file. The `oidc.json` is used to configure the native OIDC client subsystem.
++
+[source]
+----
+{
+    "client-id" : "myclient",                                                         <1>
+    "provider-url" : "${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm", <2>
+    "public-client" : "true",                                                         <3>
+    "principal-attribute" : "preferred_username",                                     <4>
+    "ssl-required" : "EXTERNAL"                                                       <5>
+}
+----
+<1> This is the client we created in Keycloak.
+<2> The provider URL, which is the URL for the realm *myrealm* that we created, is specified as an environment variable. We have set its value in the helm configuration.
+<3> When *public-client* set to *true*, client credentials are not sent when communicating with the OpenID provider.
+<4> We specify that the user name of the identity, which in our case is *alice*, is to be used as the principal for the identity.
+<5> When *ssl-required* is set to *EXTERNAL*, only the communication with external clients happens over HTTPs
+
+== Set up access to the application
+
+Set up access to your application by configuring port-forward in kubernetes.
+
+. Get the name of pod for your application:
++
+----
+kubectl get pod
+
+NAME                        READY   STATUS    RESTARTS      AGE
+keycloak-65766c8d6b-tdnhn   1/1     Running   1 (19m ago)   52m
+oidc-app-5d6f9974fd-srvrg   1/1     Running   0             4m
+----  
++
+In the example, the pod name is `oidc-app-5d6f9974fd-srvrg`.
+
+. Set port-forward:
++
+[source]
+----
+kubectl port-forward <POD_NAME> 8080:8080
+----
++
+Replace <POD_NAME> with the pod name obtained in the previous step.
+
+
+== Finish Configuring Keycloak
+
+From your *myclient* client in the Keycloak Administration Console, in the client settings, set *Valid Redirect URI* to `http://localhost:8080/simple-webapp-oidc/secured/pass:[*]` then click *Save*.
```

Comment:
These directions are more concise and clear

Click the General Settings image in the top left hand corner of the admin console
Click the "Clients" menu item
Click on "myclient" from the list on the Clients page.
Scroll down to the "Access settings" section on the page.  
In field, "Valid redirect URIs set the value to `http://localhost:8080/simple-webapp-oidc/secured/pass:[*]`
Click the "Save" button at the bottom of the page.

Reply:
agreed, updated with slight modification.